design/src/docs/pokemon_service.md

```diff
@@ -0,0 +1,239 @@
+```rust
+let request_fmt = aws_smithy_http_server::logging::sensitivity::RequestFmt::new()
+    .header(|name: &http::header::HeaderName| {
+        #[allow(unused_variables)]
+        let name = name.as_str();
+        let name_match = matches!(name, "passcode");
+        let key_suffix = None;
+        let value = name_match;
+        aws_smithy_http_server::logging::sensitivity::headers::HeaderMarker {
+            value,
+            key_suffix,
+        }
+    })
+    .label(|index: usize| matches!(index, 1));
+let response_fmt = aws_smithy_http_server::logging::sensitivity::ResponseFmt::new();
+```
+Each route is a pair, [`BoxCloneService`](https://docs.rs/tower/latest/tower/util/struct.BoxCloneService.html) wrapping the service operation (the implementation) and
+the information to consume the service operation.
+```rust
+(tower::util::BoxCloneService::new(svc), capture_pokemon_operation_request_spec)
+```
+Now the `Router` is built. `Router` is not code generated, instead lives in the aws-smithy-http-server crate; it is, whenever possible, better to write Rust code in Rust, rather than generate it to aid development.
+A `Router` is a tower `Service` that routes requests to the implemented services; hence it [implements](https://github.com/awslabs/smithy-rs/blob/db48039065bec890ef387385773b37154b555b14/rust-runtime/aws-smithy-http-server/src/routing/mod.rs#L302) `Service`
+like the other operations.
+
+To re-use code and not to overload a router with possibly unneeded operations, it also [implements](https://github.com/awslabs/smithy-rs/blob/db48039065bec890ef387385773b37154b555b14/rust-runtime/aws-smithy-http-server/src/routing/mod.rs#L146)
+the `tower::Layer` trait. Middleware are added as layers. The Pokémon example uses them:
+```rust
+let shared_state = Arc::new(State::default());
+let app = app.layer(
+    ServiceBuilder::new()
+        .layer(TraceLayer::new_for_http())
+        .layer(AddExtensionLayer::new(shared_state)),
+);
+```
+The service is run by `hyper`:
+```rust
+hyper::Server::bind(&bind).serve(app.into_make_service());
+```
+
+
+### Folder structure
+Before describing how shapes are generated, it is helpful to know the folder structure of smithy-rs.
+Code generation happens in:
+* `/codegen`: it contains shared code for both client and server, but only generates a client
+* `/codegen-server`: server only. This project started with `codegen` to generate a client, but client and server share common code; that code lives in `codegen`, which `codegen-server` depends on
+* `/aws`: the AWS SDK, it deals with AWS services specifically. The folder structure reflects the project's, with the `rust-runtime` and the `codegen`
+* `/rust-runtime`: the generated client and server crates may depend on crates in this folder. Crates here are not code generated
+
+`/rust-runtime` crates ("runtime crates") are added to a crate's dependency only when used. If a model uses event streams, it will depend on `aws-smithy-eventstream`.
```

Comment:
```suggestion
`/rust-runtime` crates ("runtime crates") are added to a crate's dependency only when used. If a model uses event streams, it will depend on [`aws-smithy-eventstream`](https://docs.rs/aws-smithy-eventstream).
```